framework/include/systems/SystemBase.h

```diff
@@ -950,6 +943,28 @@ class SystemBase : public libMesh::ParallelObject, public ConsoleStreamInterface
    */
   virtual void compute(ExecFlagType type) = 0;
 
+  /**
+   * Copy time integrators from another system
+   */
+  void copyTimeIntegrators(const SystemBase & other_sys);
+
+  /**
+   * Retrieve the time integrator that integrates the given variable's equation
+   */
+  const TimeIntegrator & getTimeIntegrator(const unsigned int var_num) const;
+
+  /**
+   * Retrieve the time integrator that integrates the given variable's equation. If no suitable time
+   * integrator is found (this could happen for instance if we're solving a non-transient problem),
+   * then a nullptr will be returned
+   */
+  const TimeIntegrator * getPossiblyNullTimeIntegrator(const unsigned int var_num) const;
+
+  /**
+   * @returns All the time integrators owned by this system
+   */
+  const std::vector<std::shared_ptr<TimeIntegrator>> & getTimeIntegrators();
```

Comment:
Does it make sense here to not grant write access? i.e., reconstruct and give a `std::vector<std::shared_ptr<const TimeIntegrator>>`

Reply:
Both `Transient` and `FEProblemBase` call non-const methods on the time integrators returned by this API. The vector should be constant, but the pointers shouldn't be